pytorch_lightning/trainer/data_loading.py

```diff
@@ -212,6 +241,19 @@ def __init__(self, num_features, dataset, *args, **kwargs):
 
         if not contains_dataset:
             dl_args.pop('dataset')
+        else:
+            # wrap the ``IterableDataset`` into a ``CaptureIterableDataset`` to record sampler states.
+            if _fault_tolerant_enabled() and isinstance(dl_args["dataset"], IterableDataset):
```

Comment:
We know that `has_iterable_dataset==True`

```suggestion
            if _fault_tolerant_enabled():
```